File: src/Samples/JS_Libs/FileLoader.fs
namespace OpenSilver.Samples.Showcase

open System
open System.Threading.Tasks
open OpenSilver

module FileLoader =
    let TryLoadJavaScriptFile (url: string) : Task<bool> =
        task {
            try
                let! _ = Interop.LoadJavaScriptFile(url)
                return true
            with
            | ex ->
                Console.WriteLine(ex)
                return false
        }
        
    let TryLoadCssFile (url: string) : Task<bool> =
        task {
            try
                let! _ = Interop.LoadCssFile(url)
                return true
            with
            | ex ->
                Console.WriteLine(ex)
                return false
        }

File: src/Samples/JS_Libs/FileLoader.vb
Namespace OpenSilver.Samples.Showcase
    Friend Module FileLoader
        Public Async Function TryLoadJavaScriptFile(url As String) As Task(Of Boolean)
            Try
                Await Interop.LoadJavaScriptFile(url)
                Return True
            Catch ex As Exception
                Console.WriteLine(ex)
                Return False
            End Try
        End Function

        Public Async Function TryLoadCssFile(url As String) As Task(Of Boolean)
            Try
                Await Interop.LoadCssFile(url)
                Return True
            Catch ex As Exception
                Console.WriteLine(ex)
                Return False
            End Try
        End Function
    End Module
End Namespace
